### DISCUSSION TITLE
How to export the complete schema as JSON or GraphQL SDL

### DISCUSSION THREAD

**Discussion by TimoFrenzel** [rebing/graphql-laravel#820] [General]
Hi all,
i work with postman.
Its possible to export the Schema as JSON or GraphQL SDL?

Greetz
Timo

**Comment by mfn** [comment]
This package does not work with SDL, so sadly I don't know how to achieve this.

But for the schema itself, yes:
- make sure `graphql.security.disable_introspection` is set to `false`
- run the query generated by `\GraphQL\Type\Introspection::getIntrospectionQuery()` against your instance
- dump the returned array as JSON

This is e.g. compatible with Apollo to generate TypeScript types, etc.

HTH!
